src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -255,6 +270,12 @@ void CARMACloudPlugin::Broadcast_TCMs()
 					event_log_msg.set_description(_TCMAcknowledgementStrategy + ": " + _TCMNOAcknowledgementDescription + " Traffic control id = " + tcmv01_req_id_hex);
 					PLOG(logDEBUG) << "event_log_msg " << event_log_msg << std::endl;
 					this->BroadcastMessage<tmx::messages::TmxEventLogMessage>(event_log_msg);	
+
+					//send negative ack to carma-cloud
+					char xml_str[10000]; 

```

Comment:
There are several code smells related your method of creating the XML payload to send to carma-cloud. There may be a better way of doing this.